app/controllers/declaration/ConfirmationController.scala

```diff
@@ -16,31 +16,101 @@
 
 package controllers.declaration
 
-import controllers.actions.AuthAction
+import scala.concurrent.{ExecutionContext, Future}
+
+import connectors.CustomsDeclareExportsConnector
+import controllers.actions.{AuthAction, VerifiedEmailAction}
+import controllers.declaration.ConfirmationController._
+import controllers.routes.{RejectedNotificationsController, SubmissionsController}
+import handlers.ErrorHandler
+import javax.inject.Inject
+import models.requests.{AuthenticatedRequest, ExportsSessionKeys}
+import play.api.Logging
 import play.api.i18n.I18nSupport
 import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
 import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
+import views.helpers.Confirmation
 import views.html.declaration.confirmation._
 
-import javax.inject.Inject
-
 class ConfirmationController @Inject()(
   authenticate: AuthAction,
+  verifyEmail: VerifiedEmailAction,
+  customsDeclareExportsConnector: CustomsDeclareExportsConnector,
   mcc: MessagesControllerComponents,
+  errorHandler: ErrorHandler,
   draftConfirmationPage: draft_confirmation_page,
   holdingConfirmationPage: holding_confirmation_page,
   submissionConfirmationPage: submission_confirmation_page
-) extends FrontendController(mcc) with I18nSupport {
+)(implicit ec: ExecutionContext) extends FrontendController(mcc) with I18nSupport with Logging {
 
   val displayDraftConfirmation: Action[AnyContent] = authenticate { implicit request =>
     Ok(draftConfirmationPage())
```

Comment:
Just wondering if we should now separate out the draft confirmation journey from the submitted declaration as they are two very different things (and I think the draft page just adds noise to this controller's main purpose IMO)